assets/js/editor.js

```diff
@@ -561,34 +563,59 @@ class Communicator {
     constructor() {}
 
     load(completion) {
-        // TODO: Actually load JSON data
-        this.uploadedData = JSON.parse(`
-        {
-          "last_submission": "fdb8da1705ac435e669120369236a8f48fdf68bc",
-          "saved_files": {
-            "test.java": "public class test {}",
-            "A.java": "public class A {}",
-            "B.java": "public class B {}",
-            "Z.java": "public class Z {}",
-            "C.java": "public class C {}"
-          }
-        }
-        `);
-        let files = [];
-        for (let key of Object.keys(this.uploadedData["saved_files"])) {
-            let file = new File(key, this.uploadedData["saved_files"][key]);
-            files.push(file);
-        }
-        completion(files, this.uploadedData["last_submission"]);
+        $.ajax({
+            url: GET_LAST_CHECKPOINT_URL,
+            type: "get",
+            headers: {"X-CSRFToken": CSRF_TOKEN},
+            datatype: "json",
+            success: function(response) {
+                if (response.success !== true) {
+                    // Return empty files and empty hash
```

Comment:
This comment is obvious and should be removed. The completion function should be explained in a central place.